prov/shm/src/smr_ep.c

```diff
@@ -303,17 +308,106 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 		    const struct iovec *iov, size_t count, size_t total_len,
 		    uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
 		    void *context, struct smr_region *smr,
-		    struct smr_resp *resp, struct smr_cmd *pend_cmd)
+		    struct smr_resp *resp, struct smr_tx_entry *pend)
 {
 	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
 	cmd->msg.hdr.op_src = smr_src_iov;
 	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) smr);
 	cmd->msg.data.iov_count = count;
 	cmd->msg.hdr.size = total_len;
-	cmd->msg.hdr.msg_id = (uint64_t) (uintptr_t) context;
 	memcpy(cmd->msg.data.iov, iov, sizeof(*iov) * count);
 
-	smr_post_pend_resp(cmd, pend_cmd, resp);
+	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
+}
+
+void smr_format_mmap(struct smr_cmd *cmd, fi_addr_t peer_id,
+		     const struct iovec *iov, size_t count, size_t total_len,
+		     uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
+		     void *context, struct smr_region *smr, uint64_t msg_id,
+		     struct smr_ep_name *map_name, void *shm_ptr,
+		     struct smr_resp *resp, struct smr_tx_entry *pend)
+{
+	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
+	cmd->msg.hdr.op_src = smr_src_mmap;
+	cmd->msg.hdr.msg_id = msg_id;
+	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) smr);
+	cmd->msg.data.iov_count = count;
+	cmd->msg.hdr.size = total_len;
+	memcpy(cmd->msg.data.iov, iov, sizeof(*iov) * count);
```

Comment:
I don't think this is necessary anymore since we're saving this directly in the tx_entry struct now